### DISCUSSION TITLE
Globalization of Navbar

### DISCUSSION THREAD

**Discussion by pallabez** [Real-Dev-Squad/website-dashboard#762] [Ideas]
# Problem
Dashboard site requires navbar for each pages to make navigation easier. Adding navbar to each html page and handling functionality would require too much repeated code. 

# Solution
## Script file to render Navbar
 - Each page that requires navbar would contain a element `<nav id="global">`, generally direct child of `<body />`.
 - Each page would also contain script `<script src="navbar.global.js" />`
 - The script fills the `<nav id="global" />` component and handle it's functionality 
 
 #### Advantage
 - Simple to implement
 - Simple to understand
 #### Disadvantage
 - Navbar re-render on page navigation
 
 ## Pseudo page navigation
 - Each page that contain a common script. 
 - The script handles page navigation via <a /> tags. 
 - On clicking a different route of the app, the default behavior is prevented.
 - The url is updated manually, and body (non navbar body) is replaced with different one. 
 
 #### Advantage
- No navbar re render
- No full page reload
#### Disadvantage 
- Harder to implement
- Harder to understand

**Comment by Achintya-Chatterjee** [comment]
@pallabez good idea
